### PR TITLE
feat(core): add agent metadata property

### PR DESCRIPTION
**Pull request by Archento**
## Proposed Changes

Add metadata property to agent constructor.
The initial version will support metadata in the following form:
```python
class AgentGeolocation(BaseModel):
    latitude: float
    longitude: float
    radius: int

class AgentMetadata(BaseModel):
    geolocation: AgentGeolocation
```

Other metadata will be ignored which means the following agent will produce the respective output:
```python
agent = Agent(
    metadata={
        "geolocation": {
            "latitude": 52.507410,
            "longitude": 13.378240,
            "radius": 50,
        },
        "mobility": True,
        "useless_information": "This is a test",
    },
)
print(agent.metadata)
>>> {'geolocation': {'latitude': 52.50741, 'longitude': 13.37824, 'radius': 50}}
```

## Types of changes

- [ ] Bug fix (non-breaking change that fixes an issue).
- [x] New feature added (non-breaking change that adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected).
- [ ] Documentation update.
- [ ] Something else (e.g., tests, scripts, example, deployment, infrastructure).

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/uAgents/blob/main/CONTRIBUTING.md) guide 
- [x] Checks and tests pass locally

### If applicable

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease
- [x] I have added/updated the documentation

### CODE REVIEW

File: python/src/uagents/agent.py [hunk 1]
```diff
@@ -508,6 +513,36 @@ def initialize_wallet_messaging(
...
+        The metadata is filtered to include only location-based metadata and the
+        model ensures that the metadata is valid and complete.
+
+        Args:
+            metadata (Dict[str, Any]): The metadata to include in the agent object.
+        """
+        if not metadata or "geolocation" not in metadata:
+            return {}
+
+        class AgentGeolocation(BaseModel):
```
Comment (Dacksus) [inline python/src/uagents/agent.py:529]: We will definitely use this set (lat, lng, radius) more often at different places in the Agent. I like the idea of making it available in the `types.py`

File: python/src/uagents/agent.py [hunk 2]
```diff
@@ -508,6 +513,36 @@ def initialize_wallet_messaging(
         else:
             self._wallet_messaging_client = None
 
+    def _initialize_metadata(self, metadata: Optional[Dict[str, Any]]):
```
Comment (jrriehl) [inline python/src/uagents/agent.py:516]:
```suggestion
    def _initialize_metadata(self, metadata: Optional[Dict[str, Any]]) -> Dict[str, Any]
```

File: python/src/uagents/agent.py [hunk 3]
```diff
@@ -508,6 +513,36 @@ def initialize_wallet_messaging(
...
+        """
+        Initialize the metadata for the agent.
+
+        The metadata is filtered to include only location-based metadata and the
+        model ensures that the metadata is valid and complete.
+
+        Args:
+            metadata (Dict[str, Any]): The metadata to include in the agent object.
+        """
```
Comment (jrriehl) [inline python/src/uagents/agent.py:525]:
```suggestion
        Args:
            metadata (Dict[str, Any]): The metadata to include in the agent object.
        Returns:
            Dict[str, Any]: The filtered metadata.
        """
```

File: python/src/uagents/agent.py (hunk 2 above)
Comment (Archento) [reply python/src/uagents/agent.py:516]: done

File: python/src/uagents/types.py [hunk 4]
```diff
@@ -55,6 +56,30 @@ class RestHandlerDetails(BaseModel):
...
+        float,
+        Field(strict=True, ge=-90, le=90, allow_inf_nan=False),
+    ]
+    longitude: Annotated[
+        float,
+        Field(strict=True, ge=-180, le=180, allow_inf_nan=False),
+    ]
+    radius: Annotated[
+        float,
+        Field(strict=True, ge=0, allow_inf_nan=False),
```
Comment (jrriehl) [inline python/src/uagents/types.py:70]: Might there be a case where infinite radius makes sense? Something entity that is available everywhere?

File: python/src/uagents/types.py [hunk 5]
```diff
@@ -55,6 +56,30 @@ class RestHandlerDetails(BaseModel):
...
+    ] = 0
+
+    @field_serializer("latitude", "longitude")
+    def serialize_precision(self, val: float) -> float:
+        """Round the latitude and longitude to 6 decimal places."""
+        return float(f"{val:.6f}")
```
Comment (jrriehl) [inline python/src/uagents/types.py:76]: What's the reason for this?

**Review by jrriehl — COMMENTED** [review]
Looks good overall! Just a minor comment and question.

File: python/src/uagents/types.py (hunk 4 above)
Comment (Archento) [reply python/src/uagents/types.py:70]: I'd be restrictive for now and open this up once we see a use-case for it.
Right now there will potentially only be local scenarios and depending on the architecture we might introduce clusters that cover broader ranges. We can get back to this at the latest when we introduce polygonal zones :)

File: python/src/uagents/types.py (hunk 5 above)
Comment (Archento) [reply python/src/uagents/types.py:76]: A precision of 6 digits after the comma equates to 10cm so I think that should be enough.
There is a funny [comparison](https://xkcd.com/2170/) from xkcd for that 😄

**Review by jrriehl — APPROVED** [review]

**Pull request merged**